superbench/benchmarks/micro_benchmarks/flops_performance.py

```diff
@@ -0,0 +1,143 @@
+        )
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+
+        self._args.precision = [p.upper() for p in self._args.precision]
+        for p in self._args.precision:
+            if p not in list(self.__kernel_map.keys()):
+                self._result.set_return_code(ReturnCode.INVALID_ARGUMENT)
+                logger.error(
+                    'Unsupported precision - benchmark: {}, precision: {}, expected: {}.'.format(
+                        self._name, p, list(self.__kernel_map.keys())
+                    )
+                )
+                return False
+            else:
+                command = os.path.join(self._args.bin_dir, self._bin_name)
+                command += (' --warmup-iterations=' + str(self._args.num_warmup))
+                command += (' --operation=gemm')
+                command += (' --n=' + str(self._args.n))
+                command += (' --k=' + str(self._args.k))
+                command += (' --m=' + str(self._args.m))
+                command += (' --kernels=' + self.__kernel_map[p])
+                self._commands.append(command)
+
+        return True
+
+    def _process_raw_result(self, cmd_idx, raw_output):
+        """Function to parse raw results and save the summarized results.
+
+          self._result.add_raw_data() and self._result.add_result() need to be called to save the results.
+
+        Args:
+            cmd_idx (int): the index of command corresponding with the raw_output.
+            raw_output (str): raw output string of the micro-benchmark.
+
+        Return:
+            True if the raw output string is valid and result can be extracted.
+        """
+        precision = self._args.precision[cmd_idx]
+        self._result.add_raw_data('raw_output_' + precision, raw_output)
+
+        flops = list()
+        content = raw_output.splitlines()
+        for line in content:
+            if 'gemm,cutlass_simt_dgemm_128x128_8x2' in line or \
+               'gemm,cutlass_simt_sgemm_128x128_8x2' in line or \
+               'gemm,cutlass_simt_hgemm_256x128_8x2' in line or \
+               'gemm,cutlass_tensorop_d884gemm_128x128_16x3' in line or \
+               'gemm,cutlass_tensorop_tf32_s1688gemm_tf32_256x128_16x3' in line or \
+               'gemm,cutlass_tensorop_bf16_s16816gemm_bf16_256x128_32x3' in line or \
+               'gemm,cutlass_tensorop_h16816gemm_256x128_32x3' in line or \
+               'gemm,cutlass_tensorop_h884gemm_256x128_32x2' in line or \
+               'gemm,cutlass_tensorop_s8_i16832gemm_s8_256x128_64x3' in line or \
+               'gemm,cutlass_tensorop_s4_i16864gemm_s4_256x128_128x3' in line:
+                flops.append(float(line.split(',')[-1]))
+
+        self._result.add_result(precision, max(flops) if len(flops) > 0 else 0)
+
```

Comment:
How to handle the error? I remember we have failure before but the code cannot catch it.

Reply:
What do you mean by "the code cannot catch it", hang? If execution of the command gets issue, the base class will handle the error.

Reply:
set return code

Reply:
Done.